app/domain/authentication/authn_azure/verify_and_decode_azure_token.rb

```diff
@@ -0,0 +1,88 @@
+require 'jwt'
+
+module Authentication
+  module AuthnAzure
+
+    Err = Errors::Authentication::AuthnAzure
+    Log = LogMessages::Authentication::AuthnAzure
+
+    # We use the OIDC log messages for Identity Provider related code
+    OIDCLog = LogMessages::Authentication::AuthnAzure
+
+    VerifyAndDecodeAzureToken = CommandClass.new(
+      dependencies: {
+        # We have a ConcurrencyLimitedCache which wraps a RateLimitedCache which wraps a FetchProviderCertificate class
+        fetch_provider_certificate: ::Util::ConcurrencyLimitedCache.new(
```

Comment:
As you wrote a comment you are probably understand that this code is not readable...
Can you break it into readable parts?

Reply:
This file is not in the PR anymore.

Reply:
And not really.